src/Executables/ReduceCceWorldtube/ReduceCceWorldtube.cpp

```diff
@@ -0,0 +1,296 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <boost/program_options.hpp>
+#include <iostream>
+#include <string>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/Cce/ReadBoundaryDataH5.hpp"
+#include "Evolution/Systems/Cce/ReducedWorldtubeModeRecorder.hpp"
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+// from a time-varies-fastest set of buffers provided by
+// `SpecWorldtubeH5BufferUpdater` extract the set of coefficients for a
+// particular time given by `buffer_time_offset` into the `time_span` size of
+// buffer.
+void slice_buffers_to_libsharp_modes(
+    const gsl::not_null<Variables<Cce::detail::cce_input_tags>*>
+        coefficients_set,
+    const Variables<Cce::detail::cce_input_tags>& coefficients_buffers,
+    const size_t time_span, const size_t buffer_time_offset, const size_t l_max,
+    const size_t computation_l_max) noexcept {
+  SpinWeighted<ComplexModalVector, 0> spin_weighted_buffer;
+
+  for (const auto& libsharp_mode :
+       Spectral::Swsh::cached_coefficients_metadata(computation_l_max)) {
+    if (libsharp_mode.l > l_max) {
+    }
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = 0; j < 3; ++j) {
+        tmpl::for_each<
+            tmpl::list<Cce::Tags::detail::SpatialMetric,
+                       Cce::Tags::detail::Dr<Cce::Tags::detail::SpatialMetric>,
+                       Tags::dt<Cce::Tags::detail::SpatialMetric>>>([
+          &i, &j, &libsharp_mode, &spin_weighted_buffer, &coefficients_buffers,
+          &coefficients_set, &l_max, &computation_l_max, &time_span, &
+          buffer_time_offset
+        ](auto tag_v) noexcept {
+          using tag = typename decltype(tag_v)::type;
+          spin_weighted_buffer.set_data_ref(
+              get<tag>(*coefficients_set).get(i, j).data(),
+              Spectral::Swsh::size_of_libsharp_coefficient_vector(
+                  computation_l_max));
+
+          Spectral::Swsh::goldberg_modes_to_libsharp_modes_single_pair(
+              libsharp_mode, make_not_null(&spin_weighted_buffer), 0,
+              libsharp_mode.l > l_max
```

Comment:
factor the ternaries out to variables outside the loops

Reply:
I don't think I can easily factor them, because the 'else' part of the ternary depends on all of the loop quantities. I'll make it an if statement so that it's more obvious what I'm doing.